api/src/main/java/fade/mirror/filter/by/Parameters.java

```diff
@@ -11,10 +11,18 @@
 
     @NotNull R withParameters(@NotNull List<Class<?>> parameterTypes, @NotNull RewriteOperation operation);
 
-    @NotNull R withParameters(@NotNull List<Class<?>> parameterTypes);
-
-    @NotNull R withParameter(@NotNull Class<?> parameterType, @NotNull RewriteOperation operation);
-
-    @NotNull R withParameter(@NotNull Class<?> parameterType);
-
+    @NotNull
```

Comment:
`@NotNull` should be placed prior to the typename, otherwise it would be a method annotation. Same with methods below.